Firestore/Source/Local/FSTLevelDBMutationQueue.mm

```diff
@@ -65,495 +69,113 @@
 using leveldb::WriteBatch;
 using leveldb::WriteOptions;
 
+static NSArray<FSTMutationBatch *> *toNSArray(const std::vector<FSTMutationBatch *> &vec) {
+  NSMutableArray<FSTMutationBatch *> *copy = [NSMutableArray array];
+  for (auto &batch : vec) {
+    [copy addObject:batch];
+  }
+  return copy;
+}
+
 @interface FSTLevelDBMutationQueue ()
 
 - (instancetype)initWithUserID:(std::string)userID
                             db:(FSTLevelDB *)db
-                    serializer:(FSTLocalSerializer *)serializer NS_DESIGNATED_INITIALIZER;
-
-/**
- * Next value to use when assigning sequential IDs to each mutation batch.
- *
- * NOTE: There can only be one FSTLevelDBMutationQueue for a given db at a time, hence it is safe
- * to track nextBatchID as an instance-level property. Should we ever relax this constraint we'll
- * need to revisit this.
- */
-@property(nonatomic, assign) BatchId nextBatchID;
-
-/** A write-through cache copy of the metadata describing the current queue. */
-@property(nonatomic, strong, nullable) FSTPBMutationQueue *metadata;
-
-@property(nonatomic, strong, readonly) FSTLocalSerializer *serializer;
+                    serializer:(FSTLocalSerializer *)serializer
+                      delegate:(std::unique_ptr<LevelDbMutationQueue>)delegate
+    NS_DESIGNATED_INITIALIZER;
 
 @end
 
 @implementation FSTLevelDBMutationQueue {
   // This instance is owned by FSTLevelDB; avoid a retain cycle.
-  __weak FSTLevelDB *_db;
+  //__weak FSTLevelDB *_db;
 
   /** The normalized userID (e.g. nil UID => @"" userID) used in our LevelDB keys. */
-  std::string _userID;
+  // std::string _userID;
```

Comment:
Can these commented out variables be simple removed?

Reply:
Yup, done.